### ISSUE TITLE
More elements in the method signature (2) than in the generic signature (1) for function ‘type<-’

### ISSUE THREAD

**Issue by bbimber** [Bioconductor/DelayedArray#101]
Hello,

We have github actions testing for our R package that includes a job that builds off bioconductor-devel. I realize this is probably not guaranteed to be able stable, but I thought i'd report this new error. This is R version 4.2.1, with bioc-devel:

```
2022-07-28T13:08:39.4926656Z DelayedArray (NA -> 0.23.0) [CRAN]
2022-07-28T13:08:39.4930098Z Installing 2 packages: DelayedArray, SummarizedExperiment
2022-07-28T13:08:39.4931077Z Summarize... (NA -> 1.27.1) [CRAN]
2022-07-28T13:08:39.4936215Z Installing packages into ‘/home/runner/work/_temp/Library’
2022-07-28T13:08:39.4937143Z (as ‘lib’ is unspecified)
2022-07-28T13:08:57.8123955Z trying URL 'https://bioconductor.org/packages/3.16/bioc/src/contrib/DelayedArray_0.23.0.tar.gz'
2022-07-28T13:08:57.9598411Z Content type 'application/x-gzip' length 642838 bytes (627 KB)
2022-07-28T13:08:58.1033762Z ==================================================
2022-07-28T13:08:58.1034226Z downloaded 627 KB
2022-07-28T13:08:58.1034456Z 
2022-07-28T13:08:58.1069951Z trying URL 'https://bioconductor.org/packages/3.16/bioc/src/contrib/SummarizedExperiment_1.27.1.tar.gz'
2022-07-28T13:08:58.2525807Z Content type 'application/x-gzip' length 1525091 bytes (1.5 MB)
2022-07-28T13:08:58.4336988Z ==================================================
2022-07-28T13:08:58.4337582Z downloaded 1.5 MB
2022-07-28T13:08:58.4337806Z 
2022-07-28T13:08:58.7737022Z 'getOption("repos")' replaces Bioconductor standard repositories, see
2022-07-28T13:08:58.7737928Z '?repositories' for details
2022-07-28T13:08:58.7738227Z 
2022-07-28T13:08:58.7738484Z replacement repositories:
2022-07-28T13:08:58.7739072Z     CRAN: https://cloud.r-project.org
2022-07-28T13:08:58.7739361Z 
2022-07-28T13:08:58.9667436Z * installing *source* package ‘DelayedArray’ ...
2022-07-28T13:08:58.9668750Z ** using staged installation
2022-07-28T13:08:58.9827432Z ** libs
2022-07-28T13:08:58.9881904Z gcc -I"/opt/R/4.2.1/lib/R/include" -DNDEBUG  -I'/home/runner/work/_temp/Library/S4Vectors/include' -I/usr/local/include   -fpic  -g -O2  -c R_init_DelayedArray.c -o R_init_DelayedArray.o
2022-07-28T13:08:59.0293519Z gcc -I"/opt/R/4.2.1/lib/R/include" -DNDEBUG  -I'/home/runner/work/_temp/Library/S4Vectors/include' -I/usr/local/include   -fpic  -g -O2  -c S4Vectors_stubs.c -o S4Vectors_stubs.o
2022-07-28T13:08:59.3957017Z gcc -I"/opt/R/4.2.1/lib/R/include" -DNDEBUG  -I'/home/runner/work/_temp/Library/S4Vectors/include' -I/usr/local/include   -fpic  -g -O2  -c abind.c -o abind.o
2022-07-28T13:08:59.4630079Z gcc -I"/opt/R/4.2.1/lib/R/include" -DNDEBUG  -I'/home/runner/work/_temp/Library/S4Vectors/include' -I/usr/local/include   -fpic  -g -O2  -c array_selection.c -o array_selection.o
2022-07-28T13:08:59.5981019Z gcc -I"/opt/R/4.2.1/lib/R/include" -DNDEBUG  -I'/home/runner/work/_temp/Library/S4Vectors/include' -I/usr/local/include   -fpic  -g -O2  -c compress_atomic_vector.c -o compress_atomic_vector.o
2022-07-28T13:08:59.6537093Z gcc -I"/opt/R/4.2.1/lib/R/include" -DNDEBUG  -I'/home/runner/work/_temp/Library/S4Vectors/include' -I/usr/local/include   -fpic  -g -O2  -c sparseMatrix_utils.c -o sparseMatrix_utils.o
2022-07-28T13:08:59.8018893Z gcc -shared -L/opt/R/4.2.1/lib/R/lib -L/usr/local/lib -o DelayedArray.so R_init_DelayedArray.o S4Vectors_stubs.o abind.o array_selection.o compress_atomic_vector.o sparseMatrix_utils.o -L/opt/R/4.2.1/lib/R/lib -lR
2022-07-28T13:08:59.8278669Z installing to /home/runner/work/_temp/Library/00LOCK-DelayedArray/00new/DelayedArray/libs
2022-07-28T13:08:59.8293834Z ** R
2022-07-28T13:08:59.8757871Z ** inst
2022-07-28T13:08:59.8788875Z ** byte-compile and prepare package for lazy loading
2022-07-28T13:09:03.8519404Z 'getOption("repos")' replaces Bioconductor standard repositories, see
2022-07-28T13:09:03.8520234Z '?repositories' for details
2022-07-28T13:09:03.8521215Z 
2022-07-28T13:09:03.8555367Z ERROR: lazy loading failed for package ‘DelayedArray’
2022-07-28T13:09:03.8556693Z * removing ‘/home/runner/work/_temp/Library/DelayedArray’
2022-07-28T13:09:03.8632555Z replacement repositories:
2022-07-28T13:09:03.8633347Z     CRAN: https://cloud.r-project.org
2022-07-28T13:09:03.8633540Z 
2022-07-28T13:09:03.8633945Z Creating a new generic function for ‘rowsum’ in package ‘DelayedArray’
2022-07-28T13:09:03.8634505Z Creating a new generic function for ‘aperm’ in package ‘DelayedArray’
2022-07-28T13:09:03.8635121Z Error in matchSignature(signature, fdef, where) : 
2022-07-28T13:09:03.8635755Z   more elements in the method signature (2) than in the generic signature (1) for function ‘type<-’
2022-07-28T13:09:03.8636638Z ##[error]Error: unable to load R code in package ‘DelayedArray’
2022-07-28T13:09:03.8637750Z Execution halted
```

**Comment by hpages** [comment]
oops, made a change in **BiocGenerics** a couple of days ago that breaks **DelayedArray**, sorry.

Should be repaired now in **DelayedArray** 0.23.1. See commit 62ca28d39ccdba6ab0029d70ae2f036e18497a70

**DelayedArray** 0.23.1 should become available via `BiocManager::install()` to BioC 3.16 users in about 24 hours.

Best

**Comment by bbimber** [comment]
ok, thanks
